### ISSUE TITLE
Server Crashes due to a server tick loop

### ISSUE THREAD

**Issue by PhuaJunLe** [hammertater/treechop#25]
Server seems to crash when trees are chopped with an item that can mine multiple blocks at once such as mekanism.

Attached a crash log.
[crash-2020-12-22_12.58.26-server.txt](https://github.com/hammertater/treechop/files/5733990/crash-2020-12-22_12.58.26-server.txt)

**Comment by hammertater** [comment]
Thanks for reporting this! You can add tools that cause crashes to a blacklist in the the server's mod config file (`[minecraft directory]/config/treechop-common.toml`). If Mekanism's atomic disassembler caused the crash, you can blacklist it by adding `"mekanism:atomic_disassembler"`:

`choppingToolsBlacklist = ["#forge:saws", "mekanism:atomic_disassembler"]`

I'll include this in the default blacklist for future versions. Eventually, I'd like to give the option to have super modded axes do multiple chops instead of breaking the whole tree (see #9).

**Comment by hammertater** [comment]
As far as fixing incompatibilities like this... the next version of this mod will include a recursion check: if a loop is detected, I'll throw a warning instead of crashing the game. This should effectively disable the chopping mechanic for most modded items that destroy multiple blocks.

**Comment by hammertater** [comment]
Version 0.11.0 is not posted, it should fix the crash without needing to modify the config. Could you close the issue if it seems to be fixed in 0.11.0?

**Comment by PhuaJunLe** [comment]
Nope doesn't seem to be fixed, although apparently one of my friends managed to join the server without the updated tree chop so maybe thats why?
[crash-2020-12-24_19.18.36-server.txt](https://github.com/hammertater/treechop/files/5739442/crash-2020-12-24_19.18.36-server.txt)
Here's the new crash report

**Comment by hammertater** [comment]
That might be the wrong crash report -- I don't see treechop installed:

```
	Mod List: 
		forge-1.16.4-35.1.13-client.jar                   |Minecraft                     |minecraft                     |1.16.4              |DONE      |NOSIGNATURE
		byg-1.1.5.jar                                     |Oh The Biomes You'll Go       |byg                           |1.1.5               |DONE      |NOSIGNATURE
		forge-1.16.4-35.1.13-universal.jar                |Forge                         |forge                         |35.1.13             |DONE      |22:af:21:d8:19:82:7f:93:94:fe:2b:ac:b7:e4:41:57:68:39:87:b1:a7:5c:c6:44:f9:25:74:21:14:f5:0d:90
		create-mc1.16.3_v0.3b.jar                         |Create                        |create                        |mc1.16.3_v0.3b      |DONE      |NOSIGNATURE
		jei-1.16.4-7.6.0.62.jar                           |Just Enough Items             |jei                           |7.6.0.62            |DONE      |NOSIGNATURE
```

Could you try to find the right report? Also, treechop needs to be updated for both the client and the server. If the client's mod is still version 0.10, the client will crash but the server should be okay.

**Comment by PhuaJunLe** [comment]
Oh sorry about that here, 
[crash-2020-12-24_11.36.10-server.txt](https://github.com/hammertater/treechop/files/5741708/crash-2020-12-24_11.36.10-server.txt)
This one shld be the correct one. 
Also thank you for being so responsive and updating so quickly ^^.

**Comment by hammertater** [comment]
No problem. Thanks for helping me find and fix bugs! The issue should definitely be fixed as of version 0.11.1 -- I just confirmed it myself. Sorry that the last "fix" didn't actually work!

In the new version, I've changed the configuration file to disable chopping when using Mekanism's atomic disassembler. So, even if the tool isn't on either of its "vein" settings, it still won't chop. You can take it off the `choppingToolsBlacklist` in the config if you want it to be able to chop, but the atomic disassembler's "vein" mode will be a little buggy on trees. It shouldn't crash though. I'd like to fix this eventually.
